### PR TITLE
Feat/ot 170 18 model migration controller category

### PR DESCRIPTION
**Pull request by DevChristians**
Campos:
id: INTEGER NOT NULL AUTO_INCREMENT
name: VARCHAR NOT NULL
description: VARCHAR NULLABLE
timestamps y softDeletes

El modelo debe tener una imagen, y se debera utilizar Active Storage

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -12,6 +12,7 @@ gem 'dotenv-rails', '~> 2.7', '>= 2.7.6' # https://rubygems.org/gems/dotenv-rail
 gem 'jsonapi-serializer', '~> 2.2' # https://rubygems.org/gems/jsonapi-serializer
 gem 'pg', '~> 1.1' # https://rubygems.org/gems/pg
 gem 'puma', '~> 5.0' # https://rubygems.org/gems/puma
+gem 'discard', '~> 1.2' #https://github.com/jhawthorn/discard
```
Comment (FerkoxGG) [inline Gemfile:15]: you need to place the gems in alphabetical order

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 Rails.application.routes.draw do
   namespace :api, defaults: { format: 'json' } do
     namespace :v1 do
+      resources :categories, only: %i[index show create update destroy]
```
Comment (FerkoxGG) [inline config/routes.rb:4]: not necessary, since it is not yet specified

File: app/models/category.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Category < ApplicationRecord
+  include Discard::Model
+  has_one_attached :image
+end
```
Comment (FerkoxGG) [inline app/models/category.rb:4]: don't forget not null value validations

**Review by FerkoxGG — COMMENTED** [review]
leave some contributions to improve

File: Gemfile (hunk 1 above)
Comment (DevChristians) [reply Gemfile]: done

File: config/routes.rb (hunk 2 above)
Comment (DevChristians) [reply config/routes.rb]: done

File: app/models/category.rb (hunk 3 above)
Comment (DevChristians) [reply app/models/category.rb:6]: done

File: app/models/category.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class Category < ApplicationRecord
+  include Discard::Model
+  has_one_attached :image
+  validates :name, presence: true
+  validates :description, presence: true
```
Comment (Ezequiel-Padilla) [inline app/models/category.rb:5]: add empty lines to separate

File: db/migrate/20220317162526_create_categories.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class CreateCategories < ActiveRecord::Migration[6.1]
+  def change
+    create_table :categories do |t|
+      t.string :name, null: false
+      t.string :description, null: false
```
Comment (Ezequiel-Padilla) [inline db/migrate/20220317162526_create_categories.rb:5]: description should be nullable

Comment (DevChristians) [reply db/migrate/20220317162526_create_categories.rb:5]: 😂 lol, sure

File: app/controllers/api/v1/categories_controller.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+      # Use callbacks to share common setup or constraints between actions.
+      def set_category
+        @category = Category.find(params[:id])
+      end
+
+      # Only allow a list of trusted parameters through.
+      def category_params
+        params.require(:category).permit(:name, :description, :image)
+      end
```
Comment (FernandoCiballero) [inline app/controllers/api/v1/categories_controller.rb:16]: Please delete these methods as they are not needed yet.

File: app/models/category.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class Category < ApplicationRecord
+  # Include Discard gem
```
Comment (FernandoCiballero) [inline app/models/category.rb:4]: In the case of Ruby and Rails it's not recommended to add comments to the code unless they are necessary.

File: app/models/category.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+  include Discard::Model
+
+  # Active Storage
```
Comment (FernandoCiballero) [inline app/models/category.rb:7]: Idem as above.

File: app/models/category.rb [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+  has_one_attached :image
+
+  # Validates
```
Comment (FernandoCiballero) [inline app/models/category.rb:10]: Idem as above.

File: spec/requests/api/v1/category_spec.rb [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    it 'returns http success' do
+      get '/api/v1/category/show'
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe 'GET /create' do
+    it 'returns http success' do
+      get '/api/v1/category/create'
+      expect(response).to have_http_status(:success)
```
Comment (FernandoCiballero) [inline spec/requests/api/v1/category_spec.rb:23]: You have to leave this file without any example, because there's not any action in the controller.

**Review by FernandoCiballero — CHANGES_REQUESTED** [review]
LGTM 🚀 left some comments.

File: app/models/category.rb (hunk 7 above)
Comment (DevChristians) [reply app/models/category.rb:4]: im working on it thanks for rview

**Review by FernandoCiballero — APPROVED** [review]
Amazing Christians!!! 🚀

**Comment by DevChristians** [comment]
hello fernando, i would like to know if the pull request is correct, thanks

**Pull request merged**